replace query-string with URLSearchParams

jmaher opened the pull request.

jmaher commented:
this was originally a renovate PR to update query-string, but query-string became an ESM module and required more work, as it was not used much, I realized we could replace it with built in URLSearchParams which reduces the need to manage as many npm packages going forward!

beatrice-acasandrei commented:
@jmaher I believe graph links from Alerts View are affected by the changes. To reproduce:
 1. Go on Alerts View (http://localhost:5000/perfherder/alerts?hideDwnToInv=1&page=1).
 2. Find some alerts.
 3. Click on the graph icon for any row, it should redirect to a graph, instead message 'Nothing here yet. Add test data to plot graphs.' is displayed.

esanuandra commented on ui/perfherder/graphs/GraphsView.jsx:
```diff
@@ -94,7 +93,7 @@ class GraphsView extends React.Component {
       highlightChangelogData,
       highlightedRevisions,
-    } = queryString.parse(this.props.location.search);
+    } = new URLSearchParams(this.props.location.search).getAll();
```
I get a console error for this line. `getAll()` seems to require an argument.
![image](https://user-images.githubusercontent.com/63001299/210561923-ce6a813e-4bd7-44bd-ba76-d6942ed25903.png)
![image](https://user-images.githubusercontent.com/63001299/210562230-58bbbd9b-f263-4233-8369-0298410d5ebc.png)

jmaher commented:
fixed the overall issue (thanks for calling it out).  clicked around a lot, added graphs, etc. and all seemed ok.  I do not have sheriff permissions to test that out.  One thing here is I have a temporary hack in the `.circleci/config.yml` file, that will go away (ideally with https://github.com/mozilla/treeherder/pull/7614), I wanted the automation to be green to not confuse things.  

please review or provide other feedback!

esanuandra commented:
hey @jmaher there is a problem with highlighted revisions
STR:
1. Go to compare view and select a comparison
2. Open graph link from a row
3. The alerts are put together with a comma in the first input
So if we have two alerts to highlight they are not separated
localhost 
![image](https://user-images.githubusercontent.com/63001299/211014013-94137665-4eaf-42c3-91e1-5c8a39632939.png)
treeherder
![image](https://user-images.githubusercontent.com/63001299/211014043-24e22785-7481-4c91-a0b4-e1486f595a99.png)

esanuandra commented:
I think we found another bug
STR:
1. Open Graphs View with some data
2. Open Job View for a data point
3. Details Panel does not load (there is an error in the console)

This happens only if you open jobs view from Graphs View.

jmaher commented on ui/models/perfSeries.js:
```diff
@@ -77,6 +75,18 @@ export const getSeriesSummary = function getSeriesSummary(
+          .map((nval) => [key, nval]),
+  );
+  return map[0];
+}
```
this is used to work around id=1,2,3,4 and instead have id=1&id=2&id=3...

jmaher commented:
thanks for finding the issue in the console.  I have tested adding graphs, compare view, clicking on data points, highlighting alerts, etc.  I do not have sheriff permissions to test anythingelse.

beatrice-acasandrei commented:
![TypeError](https://user-images.githubusercontent.com/69891317/212327958-51b8f979-fe13-455c-aa90-450e9797b1e4.PNG)
The Graphs View error is fixed but I get this error when accessing Jobs View from a datapoint in Graphs View.

jmaher commented:
I have added a small tweak and things are again looking great.  I tried searching through the code for references to using /performance/* api calls as a way to test.

jmaher commented:
this is the way the production treeherder works today:
https://treeherder.mozilla.org/perfherder/graphs?series=mozilla-central,2242563,1,4,mozilla-central,2242573,1,4,mozilla-central,2242583,1,4,mozilla-central,2242593,1,4&highlightAlerts=1&highlightCommonAlerts=0&highlightChangelogData=1&timerange=1209600

alexandru-io commented:
> this is the way the production treeherder works today: https://treeherder.mozilla.org/perfherder/graphs?series=mozilla-central,2242563,1,4,mozilla-central,2242573,1,4,mozilla-central,2242583,1,4,mozilla-central,2242593,1,4&highlightAlerts=1&highlightCommonAlerts=0&highlightChangelogData=1&timerange=1209600

interesting. i'm getting series joined by `&` you get joined by `,`.
Try: https://treeherder.mozilla.org/perfherder/graphs?series=mozilla-central,2242563,1,4&series=mozilla-central,2242573,1,4&series=mozilla-central,2242583,1,4&series=mozilla-central,2242593,1,4&highlightAlerts=1&highlightCommonAlerts=0&highlightChangelogData=1&timerange=1209600
How did you get your link?I',m just going to graphs and add all the signature by hand (Add test data)
[browsertime example](https://treeherder.mozilla.org/perfherder/graphs?highlightAlerts=1&highlightChangelogData=1&highlightCommonAlerts=0&series=autoland,3771858,1,13&series=autoland,3776763,1,13&series=autoland,3776779,1,13&timerange=1209600)

jmaher commented:
I think the missing & were a result of adding the graphs one by one with the patch.  Using the URL you provided, I was able to reproduce.  There is a quick and simple fix for this which ensures series uses .getAll()

jmaher commented:
this should be ready for testing again and review

alexandru-io commented:
> this should be ready for testing again and review

the issue is with ![image](https://user-images.githubusercontent.com/47977085/213452493-306cbccf-2a82-4e6c-a281-90fbd0e53139.png) modal. that doesn't generate correctly the URL. 
series=mozilla-central,2242563,1,4,mozilla-central,2242573,1,4,mozilla-central,2242583,1,4,mozilla-central,2242593,1,4
instead of
series=mozilla-central,2242563,1,4 &series=mozilla-central,2242573,1,4&series=mozilla-central,2242583,1,4&series=mozilla-central,2242593,1,4

beatrice-acasandrei commented:
@jmaher @alexandru-io On my side these fields are missing:
![Inkedcapture_LI](https://user-images.githubusercontent.com/69891317/215448315-7b1931d1-175c-468f-8302-ede3326ca93b.jpg)

To reproduce go to Alerts View and click on the graph icon from a random alert.

beatrice-acasandrei commented:
> hey @jmaher there is a problem with highlighted revisions STR:
> 
>     1. Go to compare view and select a comparison
> 
>     2. Open graph link from a row
> 
>     3. The alerts are put together with a comma in the first input
>        So if we have two alerts to highlight they are not separated
>        localhost
>        ![image](https://user-images.githubusercontent.com/63001299/211014013-94137665-4eaf-42c3-91e1-5c8a39632939.png)
>        treeherder
>        ![image](https://user-images.githubusercontent.com/63001299/211014043-24e22785-7481-4c91-a0b4-e1486f595a99.png)

@jmaher This is still reproducing on my side.

alexandru-io commented:
@jmaher if you want this to go faster, you can push it to prototype so we can test and share STRs and links directly from there, have the same testing environment.
